tests/Integration/Behaviour/Features/Context/Domain/OrderFeatureContext.php

```diff
@@ -1065,14 +1079,49 @@ public function orderContainsRefundedProductWithReference(string $orderReference
 
     /**
      * @Then product :productName in order :orderReference has following details:
+     * @Then combination :combinationName of product :productName in order :orderReference has following details:
      *
      * @param string $orderReference
      * @param string $productName
      * @param TableNode $table
+     * @param string|null $combinationName
+     */
+    public function checkProductDetailsWithReference(string $orderReference, string $productName, TableNode $table, ?string $combinationName = null)
+    {
+        $this->checkOrderDetailProductDetailsWithReference('first', $orderReference, $productName, $table, $combinationName);
+    }
+
+    /**
+     * @Then the :orderDetailPosition orderDetail for product :productName in order :orderReference has following details:
+     *
+     * @param string $orderDetailPosition
+     * @param string $orderReference
+     * @param string $productName
+     * @param TableNode $table
+     * @param string|null $combinationName
      */
-    public function checkProductDetailsWithReference(string $orderReference, string $productName, TableNode $table)
+    public function checkSpecificOrderDetailProductDetailsWithReference(string $orderDetailPosition, string $orderReference, string $productName, TableNode $table, ?string $combinationName = null)
     {
-        $productOrderDetail = $this->getOrderDetailFromOrder($productName, $orderReference);
+        $this->checkOrderDetailProductDetailsWithReference($orderDetailPosition, $orderReference, $productName, $table, $combinationName);
+    }
+
+    private function checkOrderDetailProductDetailsWithReference(string $orderDetailPosition, string $orderReference, string $productName, TableNode $table, ?string $combinationName = null)
+    {
+        $productOrderDetails = $this->getOrderDetailsFromOrder($productName, $orderReference, $combinationName);
+        $orderDetailsIndexes = [
+            'first' => 0,
+            'second' => 1,
+            'third' => 2,
+            'fourth' => 3,
+        ];
+
+        if (!isset($orderDetailsIndexes[$orderDetailPosition])) {
+            throw new RuntimeException(sprintf('Cannot interpret this orderDetail position %s', $orderDetailPosition));
```

Comment:
Nice-to-have: see if logic in https://github.com/PrestaShop/PrestaShop/blob/develop/tests/Integration/Behaviour/Features/Context/Util/PrimitiveUtils.php#L234 can be reused or improved